services/apps/alcs/src/portal/inbox/application/inbox-application.service.spec.ts

```diff
@@ -72,16 +98,26 @@ describe('InboxApplicationService', () => {
     );
 
     expect(result).toEqual({ data: [], total: 0 });
+    expect(mockApplicationRepository.createQueryBuilder).toHaveBeenCalledTimes(
+      1,
+    );
     expect(
-      mockApplicationSubmissionSearchViewRepository.createQueryBuilder,
-    ).toBeCalledTimes(1);
-    expect(mockQuery.andWhere).toBeCalledTimes(7);
+      mockApplicationSubmissionRepository.createQueryBuilder,
+    ).toHaveBeenCalledTimes(3);
+    expect(mockQuery.andWhere).toHaveBeenCalledTimes(3);
   });
 
-  it('should call compileApplicationSearchQuery method correctly', async () => {
-    const compileApplicationSearchQuerySpy = jest
-      .spyOn(service as any, 'compileApplicationSearchQuery')
-      .mockResolvedValue(mockQuery);
+  it('should call searchForFileNumbers method correctly', async () => {
+    mockApplicationSubmissionSearchViewRepository.createQueryBuilder.mockReturnValue(
+      mockQuery,
+    );
+
+    const searchForFileNumbersSpy = jest
+      .spyOn(service as any, 'searchForFileNumbers')
```

Comment:
Just curious: When do we use `.spyOn` and when do we just use the default mock?

Reply:
We are using spyOn here since we want to mock a single method on the class we area testing. spyOn is used for mocking a single method, createMock is when you are mocking the entire class.